test/mysql-cdc/25-schema-changes.td

```diff
@@ -92,6 +125,28 @@ contains:incompatible schema change
 
 > DROP SOURCE mz_source;
 
+#
+# ALTER TABLE <tbl> MODIFY <col> TYPE NON NULLABLE
+# Adding a non-null constraint should be allowed
+#
+
+$ mysql-execute name=mysql
+CREATE TABLE alter_column (f1 INTEGER, f2 VARCHAR(2));
+INSERT INTO alter_column VALUES (2, '12');
+
+> CREATE SOURCE mz_source FROM MYSQL CONNECTION mysql_conn
+  FOR TABLES (public.alter_column);
+
+$ mysql-execute name=mysql
+ALTER TABLE alter_column MODIFY f2 VARCHAR(2) NOT NULL;
+INSERT INTO alter_column VALUES (3, '23');
+
+> SELECT * from alter_column;
+2 12
+3 23
+
+> DROP SOURCE mz_source;
+
```

Comment:
Can we add a few more tests here? It'd be nice to have tests that will fail if we mess up correctness in the future. For example:
* Removing NOT NULL
* Changing the key

Reply:
Yep, adding tests to confirm removing NOT NULL constraints, and adding/removing unique constraints / keys